integration-tests/stateful-pallet-storage/handleItemized.test.ts

```diff
@@ -0,0 +1,179 @@
+        assert.notEqual(chainEvents["system.ExtrinsicSuccess"], undefined, "should have returned an ExtrinsicSuccess event");
+        assert.notEqual(msaCreatedEvent, undefined, "should have returned  an MsaCreated event");
+        assert.notEqual(chainEvents["transactionPayment.TransactionFeePaid"], undefined, "should have returned a TransactionFeePaid event");
+        if (msaCreatedEvent && ExtrinsicHelper.api.events.msa.MsaCreated.is(msaCreatedEvent)) {
+            msa_id = msaCreatedEvent.data.msaId;
+        }
+        assert.notEqual(msa_id, undefined, "setup should populate msa_id");
+        
+        // Grant delegation to the provider
+        const payload = await generateDelegationPayload({
+            authorizedMsaId: providerId,
+            schemaIds: [schemaId],
+        });
+        const addProviderData = ExtrinsicHelper.api.registry.createType("PalletMsaAddProvider", payload);
+
+        const grantDelegationOp = ExtrinsicHelper.grantDelegation(keys, providerKeys, signPayloadSr25519(keys, addProviderData), payload);
+        await grantDelegationOp.fundOperation();
+        const [grantDelegationEvent] = await grantDelegationOp.signAndSend();
+        assert.notEqual(grantDelegationEvent, undefined, "should have returned DelegationGranted event");
+        if (grantDelegationEvent && grantDelegationOp.api.events.msa.DelegationGranted.is(grantDelegationEvent)) {
+            assert.deepEqual(grantDelegationEvent.data.providerId, providerId, 'provider IDs should match');
+            assert.deepEqual(grantDelegationEvent.data.delegatorId, msa_id, 'delegator IDs should match');
+        }
+    });
+
+    describe("Itemized Storage Tests 😊/😥", () => {
+
+        it("✅ should be able to call applyItemizedAction and apply actions", async function () {
+            
+            // Add and update actions
+            const payload_1 = {
+                "message": "Hello World",
+            }
+            const add_action = {
+                "Add" : payload_1
+            }
+
+            const payload_2 =  {
+                "message": "Hello World Again",
+            }
+
+            const update_action = {
+                "Add" : payload_2
+            }
+
+            let add_actions = [add_action, update_action];
+            let itemized_add_result_1 = ExtrinsicHelper.applyItemActions(providerKeys, schemaId, msa_id, add_actions);
+            await itemized_add_result_1.fundOperation();
+            const [pageUpdateEvent1, chainEvents ] = await itemized_add_result_1.signAndSend();
+            assert.notEqual(chainEvents["system.ExtrinsicSuccess"], undefined, "should have returned an ExtrinsicSuccess event");
+            assert.notEqual(chainEvents["transactionPayment.TransactionFeePaid"], undefined, "should have returned a TransactionFeePaid event");
+            assert.notEqual(pageUpdateEvent1, undefined, "should have returned a PalletStatefulStorageItemizedActionApplied event");
+            // Delete action
```

Comment:
Maybe break this next section out into a separate test block? I think single-/limited-purpose tests are good style, and easier to quickly see what failed.